### DISCUSSION TITLE
在 OPNsense 上安装 tun2socks 服务 · TurboKre的博客 | Setting up tun2socks service on OPNsense · TurboKre's Blog

### DISCUSSION THREAD

**Discussion by giscus** [ohmykreee/kreee-blog#15] [Comments]
# 在 OPNsense 上安装 tun2socks 服务

A Kreee's Blog, powered by GitHub Pages, Hugo and Bilberry Premium Theme for Hugo.

https://blog.kre3.net/article/setup-tun2socks-in-opnsense/

**Comment by no1xiansen** [comment]
# tun2socks -config config.yaml
tun2socks: Command not found.
 是什么情况

**Reply by ohmykreee** [reply]
如果是严格按照博客文章内的内容进行的话，依次执行以下命令：
```bash
cd /usr/local/tun2socks
./tun2socks -config ./config.yaml
```
测试没有问题后，使用 Ctrl+C 停止当前程序。

已经修正相关内容的描述。

**Reply by no1xiansen** [reply]
非常感谢回复，现在正常运行成功了，作为小白，希望能够加上增加执行权限和下载的可执行文件需要修改名称这两个，我试了直接上传命令会执行不了，需要修改成tun2socks。再次感谢你的教程！

**Comment by no1xiansen** [comment]
希望能多出一点opnsense的教程，网上这方面的教程非常少  :smile:

**Comment by StoneMoe** [comment]
很棒的一篇笔记，也很适合作为opnsense插件开发快速概览来阅读 :D

其中插件代码的部分有一点歧义，plugins.inc.d中的inc文件添加后，应该前往`System -> Diagnostics ->Services`页面中查看是否存在 `tun-socks`，而不是`Services -> tun-socks`

**Comment by ohmykreee** [comment]
新文章 [使用 WireGuard 在 OPNsense 实现透明代理（进阶） - Kreee's Blog](https://blog.kre3.net/article/tproxy-in-opnsense-with-wireguard/)，关于使用 WireGuard 代替这里面 tun2socks 的方法，很有挑战，但是我（半知半懂）跑通了，遂发gayhub空间分享，望请各位须知 (?)
